### PR TITLE
initial rewrite of readme and contributing

### PR DESCRIPTION
**Pull request by sympatheticmoose**
Closes: https://github.com/weaveworks/weave-gitops/issues/1816

**What changed?**
Updates readme to be more user-focussed
Moves development guide under Contributing.md. NOTE: this will require follow up work to tidy-up which I am not intending to address in this PR as feel this can be tackled soon after. My focus is on the readme itself.

**Why?**

**How did you test it?**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,15 +1,149 @@
...
+6. Start the in-cluster API replacement job (powered by [http://tilt.dev](tilt.dev)) with `make cluster-dev`
+7. `make` or `make unit-tests` to ensure everything built correctly.
+8. Navigate to http://localhost:9001 in your browser. The login is `dev` with the password `dev`.
+
+### To use a bootstrapped, ready made environment
+
+1. Install go v1.17
+2. Install [buf](https://github.com/bufbuild/buf)
+3. Run `make all` to install dependencies and build binaries and assets
+4. Run `make cluster-dev` which should install and bring up everything and then start `tilt` to take over monitoring
```
Comment (Callisto13) [inline CONTRIBUTING.md:52]: I don't this is useful and what does "bootstrapped, ready made environment" mean? It's a bit misleading, as it sounds like it give you _more_ than the above flow, but it gives you way less.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,15 +1,149 @@
...
+
+### Cluster Dev Tips
+
+- You may need to turn off your `kustomize-controller` to prevent it from reconciling your "GitOps RunTime" and over-writing the `wego-app` deployment.
+- Setting the system kustomization to `suspend: true` in the config repo will also keep `kustomize-controller` from fighting with `tilt`. You may need to kill a failing pod after suspending the kustomization.
```
Comment (Callisto13) [inline CONTRIBUTING.md:57]: I am not sure what this section is for, i have never seen or used this, but that's just me 🤷‍♀️

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,15 +1,149 @@
...
+And some tools that are installed by the `tools/download-deps.sh` script:
+
+* [envtest](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/envtest) -- Run a kubernetes control plane locally for testing.
+* [tilt](https://tilt.dev/) -- Automatically build and deploy to a local cluster.
+
+## CLI/API development
+
+To set up a development environment for the gitops CLI
+
+### To use an existing environment
```
Comment (Callisto13) [inline CONTRIBUTING.md:36]: we could tie into https://github.com/weaveworks/weave-gitops/pull/1804, for a quick turnaround

**Comment by sympatheticmoose** [comment]
@Callisto13 thanks for the comments on contributing - they're good to capture 👍 

but as mentioned:
> Moves development guide under Contributing.md. NOTE: this will require follow up work to tidy-up.

I don't intend to address in this PR

**Comment by Callisto13** [comment]
>but as mentioned:

>>Moves development guide under Contributing.md. NOTE: this will require follow up work to tidy-up.

>I don't intend to address in this PR

For some weird reason I read that completely differently 😂

**Comment by sympatheticmoose** [comment]
Apologies for not making it clearer!

**Comment by Callisto13** [comment]
Looking back it is totally clear, I think my brain just fell out

**Comment by sympatheticmoose** [comment]
Closing as significantly out of date now

**Pull request closed without merging**